plasmapy/physics/transport.py

```diff
@@ -201,7 +205,1511 @@ def Coulomb_logarithm(T, n_e, particles, V=None):
+           Fokker–Planck equation." The Physics of fluids 29.4 (1986):
+           1029-1041.
+    .. [5] Ji, Jeong-Young, and Eric D. Held. "Closure and transport theory for
+           high-collisionality electron-ion plasmas." Physics of Plasmas 20.4
+           (2013): 042114.
+    """
+
+    @utils.check_quantity({"T_e": {"units": units.K, "can_be_negative": False},
+                           "n_e": {"units": units.m**-3},
+                           "T_i": {"units": units.K, "can_be_negative": False},
+                           "n_i": {"units": units.m**-3},
+                           })
+    def __init__(self, T_e, n_e, T_i, n_i, ion_particle, m_i=None, Z=None,
+                 B=0.0 * units.T, model='Braginskii',
+                 field_orientation='parallel',
+                 coulomb_log_ei=None, V_ei=None,
+                 coulomb_log_ii=None, V_ii=None,
+                 hall_e=None, hall_i=None,
+                 mu=None, theta=None):
+        # check the model
+        self.model = model.lower()  # string inputs should be case insensitive
+        valid_models = ['braginskii',
+                        'spitzer',
+                        'spitzer-harm',
+                        'ji-held',
+                        ]
+        is_valid_model = self.model in valid_models
+        if not is_valid_model:
+            raise ValueError(f"Unknown transport model '{self.model}'")
+
+        # check the field orientation
+        self.field_orientation = field_orientation.lower()
+        valid_fields = ['parallel', 'par',
+                        'perpendicular', 'perp',
+                        'cross',
+                        'all',
+                        ]
+        is_valid_field = self.field_orientation in valid_fields
+        if not is_valid_field:
+            raise ValueError(
+                f"Unknown field orientation '{self.field_orientation}'")
+
+        # density and temperature units have already been checked by decorator
+        # so just convert
+        self.T_e = T_e.to(units.K, equivalencies=units.temperature_energy())
+        self.T_i = T_i.to(units.K, equivalencies=units.temperature_energy())
+        self.n_e = n_e.to(units.m**-3)
+        self.n_i = n_i.to(units.m**-3)
+
+        # get ion mass and charge state
+        if m_i is None:
+            try:
+                self.m_i = atomic.ion_mass(ion_particle)
+            except Exception:
+                raise ValueError("Unable to find mass of particle: " +
+                                 str(ion_particle) + " in classical_transport")
+        else:
+            self.m_i = m_i.to(units.kg)
+        if Z is None:
+            try:
+                self.Z = atomic.charge_state(ion_particle)
+            except Exception:
+                raise ValueError(
+                    "Unable to find charge of particle: " +
+                    str(ion_particle) +
+                    " in classical_transport.")
```

Comment:
There are a couple of places around here where it would be great to use f-strings.  An example:
```Python
raise ValueError(f"Unable to find charge of particle {ion_particle} in classical_transport.")
```

Reply:
All hail the f-string!

Reply:
Done.